Ray moves to mainnet \o/ (#175)
golemfactory/golem-docs: src/pages/docs/creators/ray/mainnet.md

type: guide 
---

# Ray on Golem mainnet guide

The most common way of starting your adventure with Ray on Golem is by trying it out on the Golem testnet.
The testnet nodes are not very powerful and there are not many of them. 
It is enough to see how everything works and test your Ray application on.

When your application is getting more complex, and you need more power, you should move to the mainnet - there are more nodes, and the nodes themselves are more powerful, but you have to pay to use them.

There are a couple of considerations when entering the mainnet:
- [Mainnet in the cluster yaml](#mainnet-in-the-cluster-yaml) - switching to the mainnet is as simple as changing on property in the cluster yaml (besides getting funds)
- [Payments](#mainnet-payments) - operating on the mainnet might be cheap, but you still need to pay the providers running your Ray on Golem cluster
- [Funds onboarding](#mainnet-funds-onboarding) - when you don't have mainnet funds, Ray on Golem will guide you to get them
- [Budget Control](#budget-control) - Ray on Golem provides a couple of mechanisms to let you control how much you spend

> - [Finding providers](#finding-providers) - the Golem Network mainnet is peer-to-peer, meaning that the more nodes there are the more time it is required for information about them to reach you

Author: fair enough - I am removing it :heart:

If you don't have any funds, `ray up` will fail and direct you to the onboarding portal which will guide you to getting them.

Please consider using [budget control](#budget-control) mechanisms to manage your spending.

## Mainnet payments

Payments within the Golem Network take place on the [Polygon](https://polygon.technology) blockchain and require some GLM tokens (to pay the providers) and MATIC (for payment transaction fees).

Feel free to learn more about general [mainnet payments and funding your Golem node](/docs/creators/javascript/guides/switching-to-mainnet).

Note that the funds you spend on your Ray on Golem cluster don't go to Golem Factory. The GLMs go directly to providers you rent the nodes from and the MATICs go directly to Polygon blockchain validators who enable your payments.

## Mainnet funds onboarding

When your cluster yaml sports `network: "polygon"` your `ray up` attempts to run the cluster on the Golem mainnet.

If you don't have the mainnet funds, it will fail and print a link to our onboarding portal.

```bash
ray up golem-cluster.yaml --yes
```
```
(todo: ray up output with link to onboarding portal)
```

> The onboarding link is also printed in case you have the mainnet funds (it won't cause `ray up` to fail in this case) - so you can always top up your wallet.

Reviewer: ```suggestion
Even if you do have the required funds, `ray up` will nevertheless print the onboarding link as a reminder, in case you'd like to top up your wallet.
```

> The idea behind the onboarding is that when you first start Ray on Golem it generates a fresh (but empty) wallet for you. 
> This wallet is then used for both testnet and mainnet funds
> When you run on the testnet, Ray on Golem transparently gets the testnet funds for you from our testnet faucet.
> 
> The onboarding portal guides you through the creation of a web-browser wallet of your choice, getting funds there, and finally transferring GLMs and MATICs to the wallet used by Ray on Golem.
> 
> Of course, if you have a wallet you don't have to create it, if you have funds there, you don't have to get new ones. 
> 
> The ultimate goal is to have funds inside Ray on Golem.

Reviewer: tried to get a better flow there... 

```suggestion
The onboarding process is a convenient and effortless way of providing your Ray on Golem node with funds allowing it to run on mainnet.

Initially, your Ray on Golem node generates and later manages its own Ethereum wallet that is uses to perform transactions. This wallet starts empty and Ray on Golem is able to transparently obtain testnet funds when you run it on testnet. 

To fill it with mainnet funds, the onboarding portal guides you through the creation of a separate, web-browser wallet, getting GLM and MATIC tokens there, and finally transferring them to the wallet used by Ray on Golem.

Of course, if you already have your own Ethereum wallet with GLM and MATIC tokens, you don't have to get new ones. Just connect your own wallet and transfer enough of them to Ray on Golem. 
```

## Funds safety

> When handling real funds it is generally a good idea to [back up your Ray on Golem wallet](/docs/providers/wallet/backup). 

Reviewer: ```suggestion
When handling real funds, it is generally a good idea to [back up your Ray on Golem wallet](/docs/providers/wallet/backup). 
```

> Note that Ray on Golem doesn't fully support the flows around this yet. Consider operating with low amounts of money for the time being.

Reviewer: ```suggestion
Note that Ray on Golem doesn't directly support password-locked wallets yet. Therefore, we recommend operating with low amounts of funds for the time being.
```

> To proceed with the above instructions you will need to access the Golem node service `yagna`. 

Reviewer: ```suggestion
To proceed with the above instructions you will need to access the Golem node service (`yagna`). 
```

> It should be running in the background after the first successful `ray up`, but if it is not, you can always start it up with `yagna service run` command.

Reviewer: ```suggestion
Normally, it should be running in the background after the first successful `ray up`. Other than that, you can always start it manually with `yagna service run` command.
```

> Note that, if you lock your wallet (its private key actually), you will need to unlock it every time you start the Golem node (`ray up` starts it up if it is not running). 

Reviewer: ```suggestion
Note that, if you lock your wallet with a password, you will need `yagna` to unlock it every time it starts. Also, remember that `ray up` starts the `yagna` service if it doesn't find it running.
```

## Budget control

Ray on Golem provides a couple of mechanisms to give you control over your spending:

> - [hard limit](/docs/creators/ray/cluster-yaml#spending-hard-limit) of spending of each Ray on Golem cluster "session" (from `ray up` to `ray down`)

Reviewer: it's not a "session" really... just a new cluster each time... 

```suggestion
- [hard limit](/docs/creators/ray/cluster-yaml#spending-hard-limit) of spending on each Ray on Golem cluster (from `ray up` to `ray down`)
```
